client_admin/test/unit/test_pulp_tasks_extension.py

```diff
@@ -80,6 +89,79 @@ def test_list(self):
         self.assertTrue('No tasks found\n' not in self.recorder.lines)
         self.assertTrue('Result:           Incomplete\n' not in self.recorder.lines)
 
+    def test_list_all(self):
+        # Setup
+        data = {
+            'all': True,
+            'state': None
+        }
+
+        self.server_mock.request.return_value = (200, [])
+
+        # Test
+        self.all_tasks_section.list(**data)
+
+        # Verify
+        self.assertEqual(1, self.server_mock.request.call_count)
+        self.assertEqual('POST', self.server_mock.request.call_args[0][0])
+        search_crit = '{"criteria": {"fields": ["tags", "task_id", "state", "start_time", '\
```

Comment:
So, here's the thing... :) There aren't nearly as many examples of proper unit testing in our CLI code as in other parts of pulp, and you definitely didn't have any good examples to emulate in this module. It looks like this test, and the ones below, are testing a large stack of calls (into ``pulp.bindings`` and so on), and it looks like they may depend on the json serialization happening in a specific way with specific whitespace. That is actually rather fragile, so it would be better to mock ``pulp.bindings.tasks.TaskSearchAPI.search`` and make sure it got the arguments you expect. Let me know if it would be helpful to explain this more.